src/components/LoanCards/LoanCardController.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+	<component
+		:is="loanCardType"
+		:key="loan.id"
+		:loan="loan"
+		:items-in-basket="itemsInBasket"
+		:is-visitor="isVisitor"
+		:lend-increment-button-version="lendIncrementButtonVersion"
+		:image-enhancement-experiment-version="imageEnhancementExperimentVersion"
+		:is-favorite="isFavorite"
+		:expiring-soon-message="expiringSoonMessage"
+		:percent-raised="percentRaised"
+		:is-funded="isFunded"
+		:amount-left="amountLeft"
+		@track-interaction="trackInteraction"
+	/>
+</template>
+
+<script>
+import GridLoanCard from '@/components/LoanCards/GridLoanCard';
+import GridMicroLoanCard from '@/components/LoanCards/GridMicroLoanCard';
+import {
+	differenceInMinutes,
+	differenceInHours,
+	differenceInDays
+} from 'date-fns';
+import _get from 'lodash/get';
+import loanFavoriteMutation from '@/graphql/mutation/updateLoanFavorite.graphql';
+
+export default {
+	components: {
+		GridLoanCard,
+		GridMicroLoanCard,
+	},
+	props: {
+		loanCardType: {
+			type: String,
+			required: true,
+		},
+		enableTracking: {
+			type: Boolean,
+			default: false,
+		},
+		categoryId: {
+			type: Number,
+			default: null
+		},
+		categorySetId: {
+			type: String,
+			default: ''
+		},
+		rowNumber: {
+			type: Number,
+			default: null
+		},
+		cardNumber: {
+			type: Number,
+			default: null,
+		},
+		loan: {
+			type: Object,
+			default: () => {
+				return {
+					userProperties: {},
+					loanFundraisingInfo: {},
+					geocode: {
+						country: {},
+					},
+					image: {},
+				};
+			}
+		},
+		isVisitor: {
+			type: Boolean,
+			required: true,
+		},
+		itemsInBasket: {
+			type: Array,
+			default: () => [],
+		},
+		lendIncrementButtonVersion: {
+			type: String,
+			default: ''
+		},
+		imageEnhancementExperimentVersion: {
+			type: String,
+			default: ''
+		},
+	},
+	inject: ['apollo'],
+	computed: {
+		amountLeft() {
+			const {
+				fundedAmount,
+				reservedAmount
+			} = this.loan.loanFundraisingInfo;
+			return this.loan.loanAmount - fundedAmount - reservedAmount;
+		},
+		isFunded() {
+			return this.loan.status === 'funded' || this.amountLeft <= 0;
+		},
+		percentRaised() {
+			return (this.loan.loanAmount - this.amountLeft) / this.loan.loanAmount;
+		},
+		expiringSoonMessage() {
+			if (!this.loan.loanFundraisingInfo.isExpiringSoon) {
+				return '';
+			}
+			const days = differenceInDays(this.loan.plannedExpirationDate, Date.now());
+			if (days >= 2) {
+				return `Only ${days} days left! `;
+			}
+			const hours = differenceInHours(this.loan.plannedExpirationDate, Date.now());
+			if (hours >= 2) {
+				return `Only ${hours} hours left! `;
+			}
+			const mins = differenceInMinutes(this.loan.plannedExpirationDate, Date.now());
+			if (mins >= 2) {
+				return `Only ${mins} minutes left! `;
+			}
+			return 'Expiring now!';
```

Comment:
I think our next step (not for this pr) with all of these computed properties and even the methods would be to move them to a standard loan-display-utils.js file in our utils folder. We could then have methods for each which we could import individually and write better tests around.

Reply:
I like that idea, I'll keep thinking about that